src/Datasource/QueryInterface.php

```diff
@@ -148,9 +150,9 @@ public function find(string $finder, array $options = []);
      * $singleUser = $query->select(['id', 'username'])->first();
      * ```
      *
-     * @return \Cake\Datasource\EntityInterface|array|null the first result from the ResultSet
+     * @return mixed the first result from the ResultSet
```

Comment:
Why is this widened?  EntityInterface is within the same packaged and these 3 are the only possible return types AFAIK.

Reply:
Because of mapReduce()/etc.